### PR TITLE
Feature/connect to api

### PR DESCRIPTION
**Pull request by StefanieLoeser**

### CODE REVIEW

File: pages/_app.js [hunk 1]
```diff
@@ -1,23 +1,18 @@
...
-
   const [collectionState, setCollectionState] = useLocalStorage(
     '_collection',
-    myCollection
+    []
   );
 
   const [selectionState, setSelectionState] = useLocalStorage('_selection', []);
 
   function handleChange(id) {
```
Comment (NikRoe) [inline pages/_app.js:13]:
```suggestion
  function toggleBookmark(id) {
```

File: components/RecordFile.js [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
   return (
     <Record>
-      <Cover src={record.Cover} />
+      <Cover src={record.basic_information?.thumb} />
       <div>
         <BookmarkIcon
           type="checkbox"
           checked={record.isChecked}
-          onChange={() => onHandleChange(record.CatalogId)}
+          onChange={() => onChange(record.id)}
```
Comment (NikRoe) [inline components/RecordFile.js:11]: you might want to rename the toggling function to something more specific and thus prevent having an onChange event that calls a function called `onChange`.

```suggestion
          onChange={() => onToggleBookmark(record.id)}
```

File: components/RecordFile.js [hunk 3]
```diff
@@ -1,27 +1,27 @@
 import styled from 'styled-components';
 
-export default function RecordFile({ record, onHandleChange }) {
+export default function RecordFile({ record, onChange }) {
```
Comment (NikRoe) [inline components/RecordFile.js:3]:
```suggestion
export default function RecordFile({ record, onToggleBookmark }) {
```

File: pages/_app.js [hunk 4]
```diff
@@ -36,7 +31,7 @@ function MyApp({ Component, pageProps }) {
           onSetCollectionState={setCollectionState}
           selection={selectionState}
           onSetSelection={setSelectionState}
-          onHandleChange={handleChange}
+          onChange={handleChange}
```
Comment (NikRoe) [inline pages/_app.js:34]:
```suggestion
          onToggleBookmark={toggleBookmark}
```

File: pages/index.js [hunk 5]
```diff
@@ -1,8 +1,58 @@
...
+    return { ...file, isChecked: false };
+  });
+
+  return { props: { myDiscogsCollection, query } };
+}
+
+export default function Home({
+  collectionState,
+  onSetCollectionState,
+  onChange,
```
Comment (NikRoe) [inline pages/index.js:47]:
```suggestion
  onToggleBookmark,
```

File: pages/index.js [hunk 6]
```diff
@@ -11,24 +61,28 @@ export default function Home({ collectionState, onHandleChange }) {
...
-            key={file.CatalogId}
-            record={file}
-            onHandleChange={onHandleChange}
-          />
-        ))}
-      </Collection>
+      <Section>
+        <Collection>
+          {collectionState.map((file) => (
+            <RecordFile key={file.id} record={file} onChange={onChange} />
```
Comment (NikRoe) [inline pages/index.js:67]:
```suggestion
            <RecordFile key={file.id} record={file} onToggleBookmark={onToggleBookmark} />
```

File: pages/recordbag.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import styled from 'styled-components';
 import RecordFile from '../components/RecordFile';
 
-export default function RecordBag({ selection, onHandleChange }) {
+export default function RecordBag({ selection, onChange }) {
```
Comment (NikRoe) [inline pages/recordbag.js:4]:
```suggestion
export default function RecordBag({ selection, onToggleBookmark }) {
```

File: pages/recordbag.js [hunk 8]
```diff
@@ -14,7 +14,7 @@ export default function RecordBag({ selection, onHandleChange }) {
             <RecordFile
               key={file.CatalogId}
               record={file}
-              onHandleChange={onHandleChange}
+              onChange={onChange}
```
Comment (NikRoe) [inline pages/recordbag.js:17]:
```suggestion
              onToggleBookmark={onToggleBookmark}
```

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hey Stefanie,

looks great 👍 
Only a minor remark: 
You should rename your function to toggle bookmarks to something more descriptive/specific. For instance `toggleBookmark`. This will also prevent some confusion in your `RecordFile.js` component where you have the `onChange` event calling a function also called `onChange`

File: pages/recordbag.js (hunk 8 above)
Comment (StefanieLoeser) [reply pages/recordbag.js:17]: Thank you @NikRoe – I made the suggested change and renamed the onChange function in all files. Unfortunately I now get a runtime errror that onToggleBookmark is not a function. I double checked for typos and if the passing of the function is applied correctly everywhere. I also tried to destructure onToggleBookmark when being called within the onChange attribute in the component but that didn't help either. Also googling didn't help. Would you have an idea what might be the issue here? 
Many thanks in advance!

Bildschirmfoto 2022-10-04 um 11 17 47

Comment (NikRoe) [reply pages/recordbag.js]: I think this issue comes from line 17 in your `recordbag.js` file.
There it says: 
```js
onChange={onToggleBookmark}
```

but it should be 
```js
onToggleBookmark={onToggleBookmark}
```

Comment (StefanieLoeser) [reply pages/recordbag.js]: Ahhh super, thank you! I understand what the error was now. I had thought of the recordbag as a component when it is just a page that passes through the function as props. Works now!

**Review by mbosselmann — APPROVED** [review]
Hi Stefanie,
looks good to me! 👍 Nothing else to remark that Niklas has not already mentioned. :)

**Pull request merged**